### PR TITLE
First or default

### PR DESCRIPTION
**Pull request by g4s8**
First element in iterable with fallback value if iterable is empty.

### CODE REVIEW

**Comment by 0crat** [comment]
@g4s8 I'm not managing this repo, remove the [webhook](https://github.com/yegor256/cactoos/settings/hooks) or contact me in [Slack](http://www.zerocracy.com) //cc @yegor256

**Comment by yegor256** [comment]
@g4s8 good idea, but I would add an optional param to `FirstOf`. We have a similar design in `FuncWithFallback`. I don't like two classes, which are doing very similar things.

**Comment by g4s8** [comment]
@yegor256 I see. Moved fallback to `FIrstOf` contructor param.

File: src/main/java/org/cactoos/list/FirstOf.java [hunk 1]
```diff
@@ -40,23 +42,57 @@
...
     /**
      * Ctor.
      *
-     * @param source Iterable
+     * @param src Iterable
+     * @param fbk Fallback value
      */
-    public FirstOf(final Iterable<T> source) {
-        this.source = source;
+    public FirstOf(final Iterable<T> src, final Scalar<T> fbk) {
```
Comment (yegor256) [inline src/main/java/org/cactoos/list/FirstOf.java]: @g4s8 looks good, but I feel that we're not using `Scalar` the right way here. We're losing the context. This scalar won't have any idea of why it's called and what is the situation there. I would rather use `Func<Iterable<T>, T>`. This is how that `Func` will be able to throw a more meaningful exception.

**Comment by yegor256** [comment]
@g4s8 see my comment above

**Comment by g4s8** [comment]
@yegor256 you're right, replaced `Scalar<T>` with `Func<Iterable<T>, T>`.

**Comment by yegor256** [comment]
@rultor merge

**Pull request merged**